### PR TITLE
zn_poly: init at 0.9

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

Package zn_poly.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/science/math/zn_poly/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ];
+
+  nativeBuildInputs = [
+    python2 # needed by ./configure to create the makefile
+  ];
+
+  # Tuning (either autotuning or with hand-written paramters) is possible
+  # but not implemented here.
+  # It seems buggy anyways (see homepage).
+  buildFlags = "all libzn_poly.so";
```
Comment (matthewbauer) [inline pkgs/development/libraries/science/math/zn_poly/default.nix]: The libzn_poly.so doesn't seem very portable (for instance macOS uses .dylib, Windows uses .dll). Are you sure it's needed?

File: pkgs/development/libraries/science/math/zn_poly/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  installPhase = ''
+    mkdir -p "$out/include/zn_poly"
+    mkdir -p "$out/lib"
+    ls
```
Comment (matthewbauer) [inline pkgs/development/libraries/science/math/zn_poly/default.nix]: This should be removed.

Also check to make sure "make install" doesn't work- it's okay to supply installPhase just usually a lot harder to maintain.

**Comment by matthewbauer** [comment]
Is there a package that needs "zn_poly"? It's probably okay to have in Nixpkgs but note that it's unmaintained.

http://web.maths.unsw.edu.au/~davidharvey/code/zn_poly/

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Avoid libzn_poly.so target

File: pkgs/development/libraries/science/math/zn_poly/default.nix (hunk 1 above)
Comment (timokau) [reply pkgs/development/libraries/science/math/zn_poly/default.nix]: Yes its needed, its not included in `all`.

I just checked and gentoo and the sage package both special-case darwin with `libzn_poly.dylib`. Neither has a special case for windows, although sage supports windows as far as I know.

I'll add the darwin special case, but I cannot test it.

File: pkgs/development/libraries/science/math/zn_poly/default.nix (hunk 2 above)
Comment (timokau) [reply pkgs/development/libraries/science/math/zn_poly/default.nix]: Yes, its necessary. For some reason the regular `make install` does not install some header files and the so file. I've added a comment explaining this (and removed the stray `ls`).

**Comment by timokau** [comment]
Its a dependency of `sagelib`. Thanks for reviewing.

File: pkgs/development/libraries/science/math/zn_poly/default.nix (hunk 1 above)
Comment (matthewbauer) [reply pkgs/development/libraries/science/math/zn_poly/default.nix]: Okay here is the portable way to do this I think: (see https://github.com/NixOS/nixpkgs/blob/804285f589ded484b530750dd1ec03b9052bcdac/pkgs/development/python-modules/libusb1/default.nix#L15)

```
buildFlags = "all libzn_poly${stdenv.hostPlatform.extensions.sharedLibrary}";
```

Comment (timokau) [reply pkgs/development/libraries/science/math/zn_poly/default.nix]: Great! I changed that.

It should be `targetPlatform` though, right?

**Comment by timokau** [comment]
Since github apparently doesn't allow me to mark a change request as resolved: It should be resolved. The target is still there, but more portable.

**Pull request merged**